python/grass/script/core.py

```diff
@@ -1342,11 +1359,25 @@ def find_file(name, element="cell", mapset=None, env=None):
 
     :return: parsed output of g.findfile
     """
-    if element == "raster" or element == "rast":
-        verbose(_('Element type should be "cell" and not "%s"') % element)
-        element = "cell"
+    element_translation = {
+        "rast": "cell",
+        "raster": "cell",
+        "rast3d": "grid3",
+        "raster3d": "grid3",
+        "raster_3d": "grid3",
+    }
+
+    if element in element_translation:
+        element_in = element
+        element = element_translation[element]
+        debug(
+            _('Element type should be "{element}" and not "{element_in}"').format(
+                element_in=element_in, element=element
+            ),
+            1,
+        )
```

Comment:
I don't see much value in keeping this, and I don't think we are trying to discourage using 'raster' instead of cell, so the formulation 'should' is problematic anyway, no?

Reply:
Right. And in fact, the main reason I added the translation from _raster_ to _cell_ in `register.py` was to silence the verbose message which I found confusing for users who do not know where it comes from.

That said, while the element _cell_ works in `g.findfile` - which `find_file` is a wrapper for - while _raster_ does not:
```
$ g.findfile element=cell file=elevation
name='elevation'
mapset='PERMANENT'
fullname='elevation@PERMANENT'
file='/home/ninsbl/Documents/grassdata/nc_spm_08_grass7/PERMANENT/cell/elevation'

$ g.findfile element=raster file=elevation
name=
mapset=
fullname=
file=
```
So, in a way the message is correct, that if you want to use `g.findfile` to check for a raster map, you "should" use the element _cell_ and not _raster_. So, there may be value in informing the developer that the _element_ input was changed. But with the translation of code terms to mapset elements clarified in the docstring, it may be OK to remove the message alltogether.

Reply:
Yes, I do agree that the concept of `g.findfile` is not exactly intuitive, esp. for new users. A `g.find` module with and API comparable to `g.list` may be more consistent at least.
Would it be acceptable to merge this PR as it is now (it improves at least the partial workaround in _find_file_ and the documentation) and open a new issue for a `g.find` module (or maybe library function)? I would be also fine with dropping the translation of other element types and just silencing the message...

Reply:
Yes, please merge this, I was just thinking how to address this better in the future.